optimizer/evaluator/evaluator_test.go

```diff
@@ -536,7 +539,7 @@ func (s *testEvaluatorSuite) TestDateArith(c *C) {
 			Date: ast.NewValueExpr(input),
 			DateArithInterval: ast.DateArithInterval{
 				Interval: ast.NewValueExpr(t.Interval),
-				Unit:     "day",
+				Unit:     t.Unit,
```

Comment:
The test case is used for `ADDDATE(expr,days)`.  You can use the `From` to judge the type.